### PR TITLE
[v6r20] Documentation updates and minor fixes

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES

*docs
NEW: documentation on Workflow
CHANGE: update on documentation for integration tests

*Configuration
CHANGE: for pilotCS2JSONSynchronizer: if pilotFileServer is not set, still print out the content

ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
This is only docs. Please review this now as I would like to merge this immediately.

File: docs/source/AdministratorGuide/Systems/Workflow/index.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+========
+Workflow
+========
+
+[Please see this `presentation <https://indico.in2p3.fr/event/5271/contributions/33942/attachments/27190/33434/20110512_Ching_Bon_Lam.pdf>`_ for originals of the figures reported here.]
+
+The DIRAC Workflow is not properly a DIRAC system, in the sense that it doesn't implement any DB, nor service, not agent.
```
Comment (chaen) [inline docs/source/AdministratorGuide/Systems/Workflow/index.rst]: nor agent

File: docs/source/AdministratorGuide/Systems/Workflow/index.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+One more picture looking inside the contect of a workflow:
+
+.. image:: WorkflowFramework.png
+   :alt: Workflow framework.
+   :align: center
+
+So, a workflow is made of steps, modules, and each of them may have parameters. A workflow is a container of steps. A step is a container for modules.
+Steps, and modules, are ordered. There's only a linear order possible (no DAGs). Workflows are not limited in the number of steps or modules that they can execute.
+
+The following figure contains a graphycal representation of the content of a workflow.
```
Comment (chaen) [inline docs/source/AdministratorGuide/Systems/Workflow/index.rst]: graphical

File: docs/source/DeveloperGuide/CodeTesting/index.rst [hunk 3]
```diff
@@ -501,75 +507,147 @@ How do I do that?
...
    clean
 
-   echo "*** DONE ****"
+   echo -e '***' $(date -u) "*** DONE ****\n"
+
+
+This test is VERY complete, as you can see. If you are only testing locally, it may be too much,
+but as it is it's perfect for a job running in Jenkins.
+
+At the same time, out if you are a developer you should be able to extrapolated from the above those parts that you need,
```
Comment (chaen) [inline docs/source/DeveloperGuide/CodeTesting/index.rst]: out if ?

File: docs/source/DeveloperGuide/Systems/Framework/stableconns/service.py [hunk 4]
```diff
@@ -1,15 +1,18 @@
-import types
-from DIRAC import S_OK, S_ERROR
+""" example
```
Comment (chaen) [inline docs/source/DeveloperGuide/Systems/Framework/stableconns/service.py]: Shall we put here a meaning full example of docstring, with the configuration options, that will then be generated, as discussed at the DUW ?

**Comment by fstagni** [comment]
Thanks for review. Now is it OK?

**Comment by fstagni** [comment]
Some code change added, please re-review.

File: docs/source/DeveloperGuide/Systems/Framework/stableconns/service.py [hunk 5]
```diff
@@ -1,15 +1,20 @@
-import types
-from DIRAC import S_OK, S_ERROR
+""" This is a simple example of implementation of a Ping/Pong service for executors.
+
+    This service does not any specific configuration to run.
```
Comment (chaen) [inline docs/source/DeveloperGuide/Systems/Framework/stableconns/service.py]: does not have/need ? 
I would maybe still put as an example the Port parameter of the configuration

**Pull request merged**